dali/dfu/dfuwu.cpp

```diff
@@ -233,6 +245,70 @@ StringBuffer &encodeDFUsortfield(DFUsortfield fmt,StringBuffer &str,bool incmodi
     return str.append(DFUsortfields[i].str);
 }
 
+bool DfuParseLogicalPath(const IPropertyTree *directories, const char * pLogicalPath, const char* groupName,
+                StringBuffer &folder, StringBuffer &title, StringBuffer &defaultFolder, StringBuffer &defaultReplicateFolder)
+{
+    if(!pLogicalPath || !*pLogicalPath)
+        return false;
+
+    folder.clear();
+    title.clear();
+
+    defaultFolder.clear();
+    defaultReplicateFolder.clear();
```

Comment:
style: unusual to clear a StringBuffer on entry to a function. Normally semantics are that it functions only append to StringBuffers' allowing caller to pre-pend if they want to.
(and DfuParseLogicalPath is only called in 1 place, where always blank at the moment anyway)

Reply:
Right.  Mostly copied code, wrote this as shared code, but changing the other calls is outside the scope of this PR.  But I'll change as suggested.